### ISSUE TITLE
Generic callback protocol with `ParamSpec` and `TypeVar` reports error

### ISSUE THREAD

**Issue by bryanforbes** [microsoft/pyright#2431]
**Describe the bug**
Given the following code:

```python
from typing import Protocol, TypeVar
from typing_extensions import ParamSpec

P = ParamSpec('P')
R = TypeVar('R', covariant=True)


class Callback(Protocol[P, R]):
    def __call__(self, *args: P.args, **kwargs: P.kwargs) -> R:
        ...
```

An error is reported on line 8 which says `R` should be covariant

**Expected behavior**
No error should be reported

**VS Code extension or command-line**
Are you running pyright as a VS Code extension or a command-line tool?
Command line

Which version?
1.1.178
